src/working/bst.py

```diff
@@ -0,0 +1,143 @@
+"""Python implementation of Binary Search Tree."""
+from timeit import timeit
+
+
+class BST(object):
+    """Binary Search Tree."""
+
+    def __init__(self, iterable=None):
+        """Initialize Binary Search tree."""
+        self._root = None
+        self._length = 0
+        self._rdepth = 0
+        self._ldepth = 0
+        self._depth = 0
+        self._balance = 0
+        if type(iterable) in [tuple, list]:
+            for i in iterable:
+                if type(i) in [int, float]:
+                    self.insert(i)
+                else:
+                    raise TypeError('''
+Try again with only numbers in your list or tuple.''')
+        elif type(iterable) == int:
```

Comment:
BST shouldn't be taking in single values on initialization. Only iterables or nothing.